src/models/thread.ts

```diff
@@ -656,6 +660,45 @@ export class Thread extends ReadReceipt<ThreadEmittedEvents, ThreadEventHandlerM
         }
     }
 
+    /**
+     * Workaround for servers that don't support MSC3981: if we can't see the event that the threaded
+     * read receipt is pointing to, fetch it by event ID so that we can use it in our unread calculation.
+     * This is mainly for situations where the read receipt is pointing to a relation event that we don't
+     * have because it's too far back in the scrollback for us to have paginated it and it doesn't come down
+     * with the thread events due to lack of MSC3981 support.
+     *
+     * If you are reading this in the future and MSC3981 is merged and widely supported then you, future
+     * space person, get the honour of building a large pyre to incinerate this piece of code and revel
+     * in the delicious Typescript fumes.
+     */
+    private async checkForMissingReceiptEvent(): Promise<void> {
+        if (this.isCheckingForMissingReceiptEvent) return;
```

Comment:
This feels like it might want to queue/debounce rather than drop given the async action may be looking for eventId123 when the RR has moved onto eventId456 via a gappy sync during startup